portal-impl/src/com/liferay/portal/action/SitemapAction.java

```diff
@@ -78,6 +78,8 @@ public ActionForward execute(
 
 				layoutSet = LayoutSetLocalServiceUtil.getLayoutSet(
 					groupId, privateLayout);
+
+				themeDisplay.setSiteGroupId(groupId);
```

Comment:
I think that modifying the ThemeDisplay in this way is a bad idea. This is a global object (per request) so mutating it may have effects in other applications.
Where is that group ID needed? If possible, we should pass it explicitly (even if that means adding new methods somewhere else).

Reply:
Hi @adolfopa ,
When we generate a sitemap for a given site with  http://localhost:8080/sitemap.xml?groupId=<group-id>&privateLayout=false we use the groupId stored in the ThemeDisplay to get the current languages. But as the "/sitemap" is in the company scope we can gat false records I changed the ThemeDisplay here because I believe the sitemap action command is only called by the /sitemap URL so the only use-case where this change can happen in the themeDisplay is when we want to see the sitemap of a given site.

Reply:
In any case, mutating the theme display is a bad idea. Would it be possible to modify the code so that we can provide the languages in a different way? Even passing them as arguments would not be a bad idea.

Reply:
hi @adolfopa 
reverted the original change and made it more similar to https://github.com/liferay/liferay-portal/commit/be40e33509b9cb8243997939dcfc773a08c9c19b